Imports (#197)
smartcontractkit/functions-hardhat-starter-kit: README.md

);
return Functions.encodeUint256(BigInt(myNumber.toString()));
```

> ⚠️ **Users are fully responsible for any dependencies their JavaScript source code imports. Chainlink is not responsible for any imported dependencies and provides no guarantees of the validity, availability or security of any libraries a user chooses to import. Developers are advised to fully vet any imported dependencies or avoid dependencies altogether to avoid any risks associated with a compromised library or a compromised repository from which the dependency is downloaded.**

Chainlink Functions supports importing ESM-compatible modules with are supported by Deno within the JavaScript source code. It also supports importing some NPM packages [via the `npm:` specifier](https://docs.deno.com/runtime/manual/node/npm_specifiers) and some built-in Node.js modules [via the `node:` specifier](https://docs.deno.com/runtime/manual/node/node_specifiers). Check out the [Deno documentation on importing modules](https://docs.deno.com/runtime/manual/basics/modules/) for more information or visit [deno.land/x](https://deno.land/x) to find 3rd party modules which have been built for Deno.

> The total number of imports and the size of each import are restricted. At this time, users can import a maximum of 20 direct dependencies and the total size of each imported dependency cannot be larger than 10 MB (this 10 MB size limit includes any indirect dependencies required by the target library).
> 

Reviewer: ```suggestion
The total number of imports and the size of each import are restricted: 

- You can import a maximum of 20 direct dependencies.
- The total size of each imported dependency cannot be larger than 10 MB. This 10 MB size limit includes any indirect dependencies required by the target library.

```

All other execution restrictions still apply to imported dependencies. This means the dependencies will not have access to the file system, environment variables or any other Deno permissions. If an imported library requires restricted permissions, importing the library may result in an error. Furthermore, dependencies are downloaded at runtime, meaning the time required to download a dependency is counted toward the total JavaScript source code execution time limit.

> Sometimes imported dependencies will use additional fetch requests to load additional code or resources. These fetch requests will count toward the total number of HTTP requests that the JavaScript source code is allowed to perform, and if this number of allowed fetch requests is exceeded by the imported dependencies, the import attempt will fail with an error.

Reviewer: ```suggestion
Sometimes imported dependencies use additional fetch requests to load additional code or resources. These fetch requests count toward the total number of HTTP requests that the JavaScript source code is allowed to perform. If the imported dependencies exceed this total number of allowed fetch requests, the import attempt will fail with an error.
```

```
const lodash = await import("http://cdn.skypack.dev/lodash");
const result = lodash.concat([1], 2);
return Functions.encodeString(JSON.stringify(result));
```

```
const { ethers } = await import("npm:ethers@6.9.0");
const myNumber = ethers.AbiCoder.defaultAbiCoder().decode(
  ["uint256"],
  "0x000000000000000000000000000000000000000000000000000000000000002a"
);
return Functions.encodeUint256(BigInt(myNumber.toString()));
```

> ⚠️ **Users are fully responsible for any dependencies their JavaScript source code imports. Chainlink is not responsible for any imported dependencies and provides no guarantees of the validity, availability or security of any libraries a user chooses to import. Developers are advised to fully vet any imported dependencies or avoid dependencies altogether to avoid any risks associated with a compromised library or a compromised repository from which the dependency is downloaded.**

> Chainlink Functions supports importing ESM-compatible modules with are supported by Deno within the JavaScript source code. It also supports importing some NPM packages [via the `npm:` specifier](https://docs.deno.com/runtime/manual/node/npm_specifiers) and some built-in Node.js modules [via the `node:` specifier](https://docs.deno.com/runtime/manual/node/node_specifiers). Check out the [Deno documentation on importing modules](https://docs.deno.com/runtime/manual/basics/modules/) for more information or visit [deno.land/x](https://deno.land/x) to find 3rd party modules which have been built for Deno.

Reviewer 2: nit: is Nodejs standard library a better way to indicate "built in"?  I'm divided on this so either way works.

> Chainlink Functions supports importing ESM-compatible modules with are supported by Deno within the JavaScript source code. It also supports importing some NPM packages [via the `npm:` specifier](https://docs.deno.com/runtime/manual/node/npm_specifiers) and some built-in Node.js modules [via the `node:` specifier](https://docs.deno.com/runtime/manual/node/node_specifiers). Check out the [Deno documentation on importing modules](https://docs.deno.com/runtime/manual/basics/modules/) for more information or visit [deno.land/x](https://deno.land/x) to find 3rd party modules which have been built for Deno.

Reviewer 2: This para could be broken up? Example:

You can import
1. ESM-compatible NPM packages [via the `npm:` specifier](https://docs.deno.com/runtime/manual/node/npm_specifiers)
2.   ESM-compatible built-in Node.js modules [via the `node:` specifier](https://docs.deno.com/runtime/manual/node/node_specifiers)
3. ESM-Compatible 3rd party modules which have been built for Deno from  [deno.land/x](https://deno.land/x)

Check out the [Deno documentation on importing modules](https://docs.deno.com/runtime/manual/basics/modules/) for more information

Reviewer: Yes, I agree with this, except make it an unordered (bullet point) list

The total number of imports and the size of each import are restricted:

- You can import a maximum of 20 direct dependencies.
- The total size of each imported dependency cannot be larger than 10 MB. This 10 MB size limit includes any indirect dependencies required by the target library.

All other execution restrictions still apply to imported dependencies. This means the dependencies will not have access to the file system, environment variables or any other Deno permissions. If an imported library requires restricted permissions, importing the library may result in an error. Furthermore, dependencies are downloaded at runtime, meaning the time required to download a dependency is counted toward the total JavaScript source code execution time limit.

> Sometimes imported dependencies use additional fetch requests to load additional code or resources. These fetch requests count toward the total number of HTTP requests that the JavaScript source code is allowed to perform. If the imported dependencies exceed this total number of allowed fetch requests, the import attempt will fail with an error.

Reviewer 2: Recommend we link to the docs (https://docs.chain.link/chainlink-functions/resources/service-limits) in the test "allowed to perform".  
And/or expressly use the phrase "service limits" so that it maps in the readers head to what they see in the docs.
